pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -20,33 +20,199 @@ import (
 	"k8s.io/api/core/v1"
 	k8stypes "k8s.io/apimachinery/pkg/types"
 	"k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere/vclib"
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+	"strings"
+	"errors"
+	"sync"
 )
 
 // Stores info about the kubernetes node
 type NodeInfo struct {
+	dataCenter *vclib.Datacenter
 	vm       *vclib.VirtualMachine
 	vcServer string
 }
 
 type NodeManager struct {
+	// TODO: replace map with concurrent map when k8s supports go v1.9
+
 	// Maps the VC server to VSphereInstance
 	vsphereInstanceMap map[string]*VSphereInstance
 	// Maps node name to node info.
 	nodeInfoMap map[string]*NodeInfo
+	// Maps node name to node structure
+	registeredNodes map[string]*v1.Node
+
+
+	// Mutexes
+	registeredNodesLock sync.RWMutex
+	nodeInfoLock sync.RWMutex
+}
+
+const (
+	POOL_SIZE = 8
+	QUEUE_SIZE = POOL_SIZE * 10
+)
+
+var (
+	ErrNoNodeInfo = errors.New("no node info")
+	ErrNoVSphereInstance = errors.New("no vsphere instance found")
+)
+
+func (nm *NodeManager) DiscoverNode(node *v1.Node) {
```

Comment:
@BaluDontu Replying to all your rename comments here. The node manager in itself can actually be moved to a separate package (say nodemanager). It is useful to have public methods (starting with init caps) and private methods (starting with lower case) so that we know which ones are invoked from outside the file. Some of the methods like Register/Unregister/DiscoverNode, GetNodeInfo, GetVSphereInstance etc are public so they start with init-cap.

Reply:
Then its better create a new package for it rather than not following the convention.